Compiler bug unexpected type name '_option_int'

### Describe the bug

The code:

```v
module main

fn main() {
    z := true
    a := fn [z] () ?int {
        match z {
        true { return 1}
        else { return none }
        }
    }()
    b := a or {
        println("failed")
        return
    }
    println("b: ${b}")
}

```

Compilation error:
```bash
clang: warning: '-x none' after last input file has no effect [-Wunused-command-line-argument]
/tmp/v_501/v.52238980765140330.tmp.c:12264:18: error: unexpected type name '_option_int': expected expression
        _option_int a = _option_int (*_t1) () =         __closure_create(anon_fn_1db4081b921a3aab___option_int_63, (struct _V_anon_fn_1db4081b921a3aab___option_int_63_Ctx*) memdup_uncollectable(&(struct _V_anon_fn_1db4081b921a3aab___option_int_63_Ctx){.z = z,
                        ^
/tmp/v_501/v.52238980765140330.tmp.c:12264:32: error: use of undeclared identifier '_t1'
        _option_int a = _option_int (*_t1) () =         __closure_create(anon_fn_1db4081b921a3aab___option_int_63, (struct _V_anon_fn_1db4081b921a3aab___option_int_63_Ctx*) memdup_uncollectable(&(struct _V_anon_fn_1db4081b921a3aab___option_int_63_Ctx){.z = z,
                                      ^
/tmp/v_501/v.52238980765140330.tmp.c:12266:2: error: call to undeclared function '_t1'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        _t1();
        ^
3 errors generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```


### Expected Behavior

Should compile. Below code compiles successfully.
```
module main

fn main() {
    a := fn () int {
        return 1
    }()
    println("a: ${a}")
}

```

### Current Behavior

Compilation error.

### Reproduction Steps

Run the code added to the description.

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.0 4f518c2

### Environment details (OS name and version, etc.)

ProductName:		macOS
ProductVersion:		13.4.1
BuildVersion:		       22F82

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -683,9 +683,12 @@ fn (mut g Gen) call_expr(node ast.CallExpr) {
 			tmp_var := g.new_tmp_var()
 			fn_type := g.fn_var_signature(node.left.decl.return_type, node.left.decl.params.map(it.typ),
 				tmp_var)
+			line := g.go_before_stmt(0).trim_space()
+			g.empty_line = true
 			g.write('${fn_type} = ')
 			g.expr(node.left)
 			g.writeln(';')
+			g.write(line)
 			g.write(tmp_var)
 		} else if node.or_block.kind == .absent {
 			g.expr(node.left)

--- a/vlib/v/tests/anon_fn_direct_call_with_option_test.v
+++ b/vlib/v/tests/anon_fn_direct_call_with_option_test.v
@@ -1,0 +1,17 @@
+fn test_anon_fn_direct_call_with_option() {
+	z := true
+	a := fn [z] () ?int {
+		match z {
+			true { return 1 }
+			else { return none }
+		}
+	}()
+	b := a or {
+		println('failed')
+		return
+	}
+
+	println('b: ${b}')
+	println(a)
+	assert b == 1
+}

